examples/MomentoApplication/Program.cs

```diff
@@ -30,17 +32,21 @@ async static Task Main(string[] args)
             do
             {
                 ListCachesResponse resp = client.ListCaches(token);
-                foreach (CacheInfo cacheInfo in resp.Caches)
+                if (resp is ListCachesResponse.Success successResult)
                 {
-                    Console.WriteLine(cacheInfo.Name);
+                    foreach (CacheInfo cacheInfo in successResult.Caches)
+                    {
+                        Console.WriteLine(cacheInfo.Name);
+                    }
+                    token = successResult.NextPageToken;
                 }
-                token = resp.NextPageToken;
             } while (!String.IsNullOrEmpty(token));
             Console.WriteLine($"\nSetting key: {KEY} with value: {VALUE}");
             await client.SetAsync(CACHE_NAME, KEY, VALUE);
             Console.WriteLine($"\nGet value for  key: {KEY}");
             CacheGetResponse getResponse = await client.GetAsync(CACHE_NAME, KEY);
-            Console.WriteLine($"\nLookedup value: {getResponse.String()}, Stored value: {VALUE}");
+            var hitResponse = (CacheGetResponse.Hit)getResponse;
```

Comment:
I think you probably want to do the `if` check here too, and use that same variable binding syntax that you used up on line 37